src/c++/perf_analyzer/data_loader.cc

```diff
@@ -339,43 +339,35 @@ DataLoader::GetInputData(
     // Get the data and the corresponding byte-size
     auto it = input_data_.find(key_name);
     if (it != input_data_.end()) {
+      data.is_valid = true;
+
       if (input.datatype_.compare("BYTES") != 0) {
-        *batch1_size = it->second.size();
+        data.batch1_size = it->second.size();
       } else {
         std::vector<char>* string_data;
         string_data = &it->second;
-        *batch1_size = string_data->size();
-      }
-
-
-      if (it->second.size()) {
-        *data_ptr = (const uint8_t*)&((it->second)[0]);
-      } else {
-        // If data is found but empty, we still want to return a non-null
-        // pointer. In that case, just point to the vector itself (instead of
-        // the raw data)
-        *data_ptr = (const uint8_t*)&(it->second);
+        data.batch1_size = string_data->size();
       }
 
-      data_found = true;
+      data.data_ptr = (const uint8_t*)&((it->second)[0]);
```

Comment:
Since this is a cleanup, mind if we break this line up into more so we can see what each dereference, etc is?

Reply:
fixed